apps/content_filters/content_filter/content_filter_service.py

```diff
@@ -193,34 +193,34 @@ def _get_elastic_query(self, doc, matching=True):
             expressions_list.append({"bool": filter_conditions})
         return {"bool": expressions}
 
-    def does_match(self, content_filter, article, filters=None):
+    def does_match(self, content_filter, article, filters=None, cache=True):
         if not content_filter:
             return True  # a non-existing filter matches every thing
         cache_id = _cache_id("filter-match", content_filter.get("_id") or content_filter.get("name"), article)
-        if not hasattr(flask.g, cache_id):
-            setattr(flask.g, cache_id, self._does_match(content_filter, article, filters))
+        if not cache or not hasattr(flask.g, cache_id):
+            setattr(flask.g, cache_id, self._does_match(content_filter, article, filters, cache))
```

Comment:
imo when `cache=False` it shouldn't set it neither, so rather do something like:
```py
if not cache:
    return self._does_match(content_filter, article, filters, cache)
elif not hasattr(...):
....
```

Reply:
and that should be applied everywhere